x-pack/legacy/plugins/ml/public/jobs/jobs_list/components/edit_job_flyout/edit_job_flyout.js

```diff
@@ -73,10 +77,37 @@ class EditJobFlyoutUI extends Component {
     }
   }
 
-  closeFlyout = () => {
+  closeFlyout = (isConfirmed = false) => {
+    if (this.containsUnsavedCustomUrls() && !isConfirmed) {
```

Comment:
I like this check for unsaved changes. It would be good to extend it to the other tabs in the flyout! Would this be a lot of extra work?

Reply:
Implemented in https://github.com/elastic/kibana/pull/47926/commits/138b4e5bde0aa4f4d42b0ac591caa61d678a7ba6